extras/module-node/README.rst

```diff
@@ -0,0 +1,62 @@
+=============================
+Node.js-like module framework
+=============================
+
+This directory contains an example module resolution and loading framework and
+``require`` implementation based on Node.js.  The application need only
+provide the module resolution and loading logic:
+
+* Add ``duk_module_node.c`` to list of C sources to compile.
+
+* Ensure ``duk_module_node.h`` is in the include path.
+
+* Include the extra header in calling code and initialize the bindings::
+
+      #include "duktape.h"
+      #include "duk_module_node.h"
+
+      /* After initializing the Duktape heap or when creating a new
+       * thread with a new global environment:
+       */
+      duk_module_node_init(ctx, resolve_cb, load_cb);
```

Comment:
Assuming that either (or both) of #575 or the format string-based `duk_put_props()` will eventually make it into master, then I'm fine with using an object.